### ISSUE TITLE
Sorting Community Solutions

### ISSUE THREAD

**Issue by FaisalAfroz** [exercism/exercism#5616]
It would be good if the community solutions could be sorted by 
1. Date
2.  Stars
3.  Comments.

The v2 website already had this feature.

**Comment by iHiD** [comment]
I'm pushing a fix to change this so they order by most stars and then newest. I'll leave this open to remind us to add custom sorting eventually.

**Comment by szabgab** [comment]
Another issue is that now, at least in [this solution](the https://exercism.org/tracks/bash/exercises/hamming/solutions) most of the solutions, and in particular the first 5 solutions have the triangle of expiration on them. Maybe a checkbox to show/hide solutions that are for a previous version of the exercise?

**Comment by masahx** [comment]
> 
> 
> I'm pushing a fix to change this so they order by most stars and then newest. I'll leave this open to remind us to add custom sorting eventually.

Maybe just by date and user who is watching first, like in previous version ;) Because some solution can jump to beginning if it get more stars. 
Or to have some mark seen-unseen. I mean this if you consider another options

**Comment by hraftery** [comment]
As hinted already by others, the big use case for me is looking for quality solutions to the problem I just solved. Unfortunately the first 100 pages or so are (quality) solutions to the problem as it was 3 years ago, which is not useful. My experience is with the Haskell track, and there appears to have been changes to almost every problem since the most popular solutions were posted. I'm sure those oh-so-important relevant solutions are in the list somewhere, but without better sorting I can't find them, which is frustrating.

**Comment by glennj** [comment]
Maybe we could get a button on the exercise solutions page that toggles whether outdated solutions are hidden.

**Comment by iHiD** [comment]
@ErikSchierboom's working on it as we speak. The biggest challenge is that all old solutions need processing to get their tests status set correctly against the latest version of the tests, which involves a lot of processing (running tests on millions of iterations)

**Comment by iHiD** [comment]
FYI, we hope to launch this this week. If @ErikSchierboom doesn't get pulled into too many other things then we should get it deployed :)

**Comment by ErikSchierboom** [comment]
Once https://github.com/exercism/website/pull/2189 is merged and deployed, you'll be able to:

- Sort by num stars (descendingly)
- Filter out-of-date exercises
- Filter exercises that pass their exercise's tests
- Filter exercises that pass their exercise's latest (head) tests

**Comment by hraftery** [comment]
Well done, all! Now I don't have to scroll through hundreds of pages of starred solutions to find one that's relevant to the current version of the problem. Instead I can see, as I suspected, that none of the solutions to the current problem have been starred yet ;-)

Hopefully this change enables people to apply some stars to some contemporary solutions, benefiting students to come.

**Comment by iHiD** [comment]
Thanks @hraftery - there's a few other PRs coming today that enhance all this too :)
